### ISSUE TITLE
Open via URL

### ISSUE THREAD

**Issue by qfard** [josdejong/jsoneditor#1378]
I can open a local file and also cut/paste content in to the code view and everything works fine.

But when I try to open the same file using full qualified URL (accessible via http), I get "TypeError: Failed to fetch" error.

Also, can I open the https://jsoneditoronline.org/ with URL file as a parameter to integrate it to my back-end debugging tool?

**Comment by josdejong** [comment]
Yes you can open a url via the menu "Open" > "Open from url". The url will be put in a query parameter in the url, like 

```
https://jsoneditoronline.org/#left=url.<ENCODED URL HERE>

https://jsoneditoronline.org/beta/#left=url.<ENCODED URL HERE>
```

The url needs to be public. If an url doesn't work, it is probably your browser blocks it due to security restrictions.

**Comment by qfard** [comment]
Thank you for the response.

 

Can I open the left & right with 2 different JSONs on a single call?

**Comment by josdejong** [comment]
Yes, like

```
https://jsoneditoronline.org/#left=url.<ENCODED URL HERE>&right=url.<ENCODED URL HERE>
```

**Comment by josdejong** [comment]
The example url you share is a bit broken probably due to your email application

So the two encoded urls are:
```
http%3A%2F%2Fsync.gohitsdigital.com%2FbatchFiles%2Fleft.json
http%3A%2F%2Fsync.gohitsdigital.com%2FbatchFiles%2Fright.json
``` 

And the final url for the editor:
```
https://jsoneditoronline.org/#left=url.http%3A%2F%2Fsync.gohitsdigital.com%2FbatchFiles%2Fleft.json&right=url.http%3A%2F%2Fsync.gohitsdigital.com%2FbatchFiles%2Fright.json
``` 

This indeed does not work. When you enter a url in the editor via the menu Open -> Open Url, the requirements are explained:

> Urls which need authentication or do not have CORS enabled cannot be loaded.

In your case there are two problems: 

1. the urls are unsecured `http` requests, whilst you open the secured `https` version of the website. The browser will block unsecured http requests. You can solve that by opening `http://jsoneditoronline.org`  instead of `https://jsoneditoronline.org`
2. since this is a cross domain request, the response must have a CORS header like `Access-Control-Allow-Origin: *`. Without this header your browser will block cross domain requests.

**Comment by josdejong** [comment]
CORS (not "CORES"), Cross-Origin Resource Sharing is about browser security restrictions. There are a lot of websites explaining what it is, you can do some googling about it. This CORS header must be added to the HTTPS request by the server. This is not always possible if you're not in control of the server.

**Comment by josdejong** [comment]
👍 

> I guess I have to find another tool.

yeah I guess so, or you'll have to find a different way to share your JSON files such that they are accessible. Will close this issue now.
